account_banking_payment_export/models/account_payment.py

```diff
@@ -121,24 +158,36 @@ def action_open(self):
         setting of the payment.order
         Re-generate the bank payment lines
         """
+        create_time = datetime.now()
+
         res = super(PaymentOrder, self).action_open()
-        bplo = self.env['bank.payment.line']
         today = fields.Date.context_today(self)
         for order in self:
+            date_update_dict = {}
+            if order.date_prefered == 'due':
+                for payline in order.line_ids:
+                    requested_date = payline.ml_maturity_date or today
+                    if requested_date not in date_update_dict:
```

Comment:
anything against `date_update_dict.setdefault(requested_date, []).append(payline.id)`? Pythonic FTW!

Reply:
Way to adventurous for me! But let's give it a try ;-).